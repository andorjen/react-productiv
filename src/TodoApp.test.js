import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoApp from "./TodoApp";

const testTodos = [ {
    id: 1,
    title: "Code!",
    description: "Write some code",
    priority: 2,
  },
  {
    id: 2,
    title: "Make dinner",
    description: "Cook something healthy",
    priority: 1,
  },
  {
    id: 3,
    title: "Go to bed",
    description: "In bed by 11:15",
    priority: 3,
  }];


it("renders without crashing", function () {
    render(<TodoApp initialTodos={testTodos} />);
  });

it("matches snapshot", function () {
    const { container } = render(<TodoApp initialTodos={testTodos}  />);
    expect(container).toMatchSnapshot();
});


it("renders the correct elements", function () {
    const {container, debug}= render(<TodoApp initialTodos={testTodos} />);
 
    const editableTodoList = container.querySelector('.EditableTodoList');
    expect(editableTodoList).toContainHTML("Make dinner");
    expect(editableTodoList).toContainHTML("Cook something healthy");
    expect(editableTodoList).toContainHTML("(priority: 1)");
    expect(editableTodoList).toContainHTML("Code!");
    expect(editableTodoList).toContainHTML("Write some code");
    expect(editableTodoList).toContainHTML("Go to bed");
    expect(editableTodoList).toContainHTML("In bed by 11:15");
    expect(editableTodoList).toContainHTML("Edit");
    expect(editableTodoList).toContainHTML("Del");

    const topTodo = container.querySelector('.topTodo');
    expect(topTodo).toContainHTML("<h3>Top Todo</h3>");
    expect(topTodo).toContainHTML("(priority: 1)");

    const todoForm = container.querySelector('.NewTodoForm');
    debug(todoForm);
    expect(todoForm).toContainHTML("GÃ¸");
 
  });

// it("renders the correct form fields", function () {
//     const {container, debug} = render(<TodoForm initialFormData={initialFormData} />);

//     expect(container).toContainHTML('class="NewTodoForm"');
//     expect(container).toContainHTML('name="title"');
//     expect(container).toContainHTML('name="description"');
//     expect(container).toContainHTML('name="priority"');
// });

// it("renders the correct data before editing", function () {
//     const {container, debug} = render(<TodoForm initialFormData={initialFormData} />);

//     const titleField = container.querySelector("#newTodo-title");   
//     expect(titleField.value).toEqual('');

//     const descriptionField = container.querySelector("#newTodo-description");
//     expect(descriptionField.value).toEqual('');

//     const priorityField = container.querySelector("#newTodo-priority");
//     expect(priorityField.value).toEqual("1");
// });


// it("renders the correct data after editing", function () {
//     const {container, debug} = render(<TodoForm initialFormData={initialFormData} />);

//     // fireEvent onChange?

//     //Data after changing reflects the changes made
//     const titleField = container.querySelector("#newTodo-title");   
//     expect(titleField.value).toEqual('changed title');

//     const descriptionField = container.querySelector("#newTodo-description");
//     expect(descriptionField.value).toEqual('changed description');

//     const priorityField = container.querySelector("#newTodo-priority");
//     expect(priorityField.value).toEqual("2");

// });


