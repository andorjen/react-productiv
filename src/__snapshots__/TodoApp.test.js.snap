// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <main
    class="TodoApp"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="EditableTodoList"
        >
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3 eachTodo"
            >
              <div
                class="Todo"
                id="1"
              >
                <div>
                  <b>
                    Code!
                  </b>
                   
                  <small>
                    (priority: 
                    2
                    )
                  </small>
                </div>
                <div>
                  <small>
                    Write some code
                  </small>
                </div>
              </div>
              <div
                class="float-right text-sm-right edit"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
            </div>
          </div>
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3 eachTodo"
            >
              <div
                class="Todo"
                id="2"
              >
                <div>
                  <b>
                    Make dinner
                  </b>
                   
                  <small>
                    (priority: 
                    1
                    )
                  </small>
                </div>
                <div>
                  <small>
                    Cook something healthy
                  </small>
                </div>
              </div>
              <div
                class="float-right text-sm-right edit"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
            </div>
          </div>
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3 eachTodo"
            >
              <div
                class="Todo"
                id="3"
              >
                <div>
                  <b>
                    Go to bed
                  </b>
                   
                  <small>
                    (priority: 
                    3
                    )
                  </small>
                </div>
                <div>
                  <small>
                    In bed by 11:15
                  </small>
                </div>
              </div>
              <div
                class="float-right text-sm-right edit"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <section
          class="mb-4 topTodo"
        >
          <h3>
            Top Todo
          </h3>
          <div
            class="Todo"
            id="2"
          >
            <div>
              <b>
                Make dinner
              </b>
               
              <small>
                (priority: 
                1
                )
              </small>
            </div>
            <div>
              <small>
                Cook something healthy
              </small>
            </div>
          </div>
        </section>
        <section>
          <h3
            class="mb-3"
          >
            Add Nü
          </h3>
          <form
            class="NewTodoForm"
          >
            <div
              class="form-group"
            >
              <input
                aria-label="Title"
                class="form-control"
                id="newTodo-title"
                name="title"
                placeholder="Title"
                required=""
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <textarea
                aria-label="Description"
                class="form-control"
                id="newTodo-description"
                name="description"
                placeholder="Description"
                required=""
              />
            </div>
            <div
              class="form-group d-flex justify-content-between"
            >
              <div
                class="w-75 d-flex justify-content-between"
              >
                <label
                  class="d-inline-flex"
                  for="newTodo-priority"
                >
                  Priority:  
                </label>
                <select
                  class="form-control form-control-sm d-inline-flex"
                  id="newTodo-priority"
                  name="priority"
                >
                  <option
                    value="1"
                  >
                    Ultra-Über
                  </option>
                  <option
                    value="2"
                  >
                    Über
                  </option>
                  <option
                    value="3"
                  >
                    Meh
                  </option>
                </select>
              </div>
              <button
                class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
              >
                Gø!
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
  </main>
</div>
`;
